### ISSUE TITLE
Give me all users and all of their roles...BY NAME

### ISSUE THREAD

**Issue by coscallis** [aspnet/Identity#926]
This is among the most simple of Use-Cases and yet does not appear to be allowed. When administering my site's users I need to be able to get a complete list of my users with their basic properties, one of which is the collection of roles assigned to the user. Using the  ApplicationUser and UserManager "out of the box" and this code: _(ok, yes, mgr and org represent properties which have been added to ApplicationUser, but they don't change the present case)_

`var results = UserManager.Users.Where(u => u.mgr == appUser.org || appUser.org == "corp");`

I get all of the expected users ... but their "role" property only gives me the "RoleId" (and UserId) NOT the role's name. Since the database does enforce referential integrity between AspNetUserRoles and AspNetRoles it really SHOULD be a simple matter of including role names as part of this query...but does not appear to be so.

**Comment by gdoron** [comment]
Did you try adding `.Include` to your query?
I guess you're using EF, right?

**Comment by coscallis** [comment]
Yes, Include(...) with many variations could only get me as far as the RoleId... I have received some help from [Stack Overflow](http://stackoverflow.com/a/38746024/681009) on how to custom configure Identity, and while this 'works' I find it very dissatisfactory as it is overly complicated for what should, provided the db schema that is initialized with Identity, should be a simple relationship.

**Comment by blowdart** [comment]
@divega

**Comment by Qszer94** [comment]
I'm also having a hard time trying to get collection of users with collection of their IdentityRoles.
As @coscallis said we can simply access only IdentityUserRole (RoleId, UserId) with EF.
It's a bit confusing cause we can get all Users of IdentityRole but not vice versa.

**Comment by flatproject** [comment]
Sorry for digging out this but since it's not closed I would like to ask if we will be able to get a list of users with their roles by default at any of the forthcoming versions? 
An official example of how to get a list of all users with all their role names would be great whether the above functionality requested is implemented or not.

**Comment by J0nKn1ght** [comment]
I'd add to the previous posts in reiterating that this feature is a necessity if Identity is to be usable in anything but the most trivial of applications.
We've just started a major new project, and decided to switch from MVC5 to ASP.Net Core MVC.  However, I've now started on the User Administration part of the site, and found that it's currently not possible to query the users by role.  The site developing could potentially grow to 10,000s of users, so querying users first and then stitching in their roles is not an option - it has to be done at the DB query level.
This was possible in ASP.Net Identity 2.0, so presumably someone decided not to implement the navigation properties on the base entities in Core?
As an aside, we were also attempting to follow the advice that seems to be coming out of Microsoft to use Clean Architecture (or Onion Architecture) in our application, but again the implementation of Identity makes that impossible.
Don't get me wrong, I've been consistently impressed by MVCs development over the years, but Core seems to have taken a number of steps backwards that are not immediately obvious until you try to use it for real development projects.  The tools and libraries need to be up to real-world scenarios, not just tutorials.
